### PR TITLE
Fixed Subcircuit overflowing vertically

### PR DESCRIPTION
**Pull request by ryuk156**
Fixes  https://github.com/CircuitVerse/CircuitVerse/issues/815

#### Describe the changes you have made in this PR -
When there are many subcircuits in the simulator the Insertsubcircuitdialog  overflow and make simulator broken added scrollbar and  fixed height and width to dialog 
### Screenshots of the changes (If any) -
Before:
image

After:

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by yash15601** [comment]
@satu0king  @tachyons  @nitin10s  Review please

**Comment by yash15601** [comment]
@nitin10s  i think it is not possible to have 5-6 rows because when their is one subcircuit inside it will have lot of blank space like this
image

**Comment by Shivansh2407** [comment]
@yash15601 I think the dialog box should work like, when there is one circuit available for inserting then the space should not be there and on increasing the circuits the dialog box height should increase till 5-7 rows visible at a time. Like we should be dynamically adding the rows in the dialog box till the maximum of 5-7 circuits are available and then scrolling will come into action. @nitin10s let me know whether this will be fine .

**Comment by yash15601** [comment]
@shivansh2407 why clear description? This what nitin is saying it is as simple to understand  please stop  repeating  things in pr's

**Comment by yash15601** [comment]
@nitin10s  @satu0king  @tachyons  Review

**Comment by yash15601** [comment]
@nitin10s  review  please

**Comment by yash15601** [comment]
@nitin  review  this please

File: public/js/data.js [hunk 1]
```diff
@@ -822,7 +822,10 @@ createSubCircuitPrompt = function(scope = globalScope) {
     }
     if (flag) $('#insertSubcircuitDialog').append('<p>Looks like there are no other circuits which doesn\'t have this circuit as a dependency. Create a new one!</p>')
     $('#insertSubcircuitDialog').dialog({
-        width: "auto",
+        maxHeight: 350,
+        width: 250,
```
Comment (aayush-05) [inline public/js/data.js:826]: Does the width attributes contradict each other ? I think just specifying the width would do

File: app/views/simulator/edit.html.erb [hunk 2]
```diff
@@ -212,7 +212,7 @@
         </div>
       </div>
       <div id="combinationalAnalysis" style="display:none;" title="BooleanLogicTable"></div>
-      <div id="insertSubcircuitDialog" style="display:none;" title="Insert SubCircuit"></div>
+      <div id="insertSubcircuitDialog" style="display:none;overflow-x:hidden;overflow-y:auto" title="Insert SubCircuit"></div>
```
Comment (tachyons) [inline app/views/simulator/edit.html.erb:215]: css please

Comment (yash15601) [reply app/views/simulator/edit.html.erb:215]: Their is no change in css @tachyons  the subcircuit dialog   is generated using js

Comment (Shivansh2407) [reply app/views/simulator/edit.html.erb:215]: The internal css needs to be transferred to respective external css file @yash15601.

Comment (yash15601) [reply app/views/simulator/edit.html.erb:215]: I have not made any changes  in css it was already

Comment (nitin10s) [reply app/views/simulator/edit.html.erb:215]: Please do it anyways. Thanks.

**Comment by yash15601** [comment]
Okay will make changes soon

**Comment by ryuk156** [comment]
@satu0king  review please

**Pull request merged**